File: Java/AutomationFramework/src/com/app/helpers/drivers/Chrome.java
package com.app.helpers.drivers;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;

public class Chrome implements IDriver{
    WebDriver driver;

    @Override
    public WebDriver GetDriver() {
        if(driver == null)
            SetDriver();

        return driver;
    }

    @Override
    public void SetDriver() {
        String path = System.getProperty("user.dir");
        System.setProperty("webdriver.chrome.driver", path + "\\drivers\\chromedriver.exe");
        ChromeOptions options = new ChromeOptions();
        options.addArguments("--remote-allow-origins=*");
        this.driver = new ChromeDriver(options);
    }

    @Override
    public void MaximizeDriver() {
        driver.manage().window().maximize();
    }

    @Override
    public void MinimizeDriver() {
        driver.manage().window().minimize();
    }

    @Override
    public void NavigateURL(String url) {
        driver.get(url);
    }

    @Override
    public void CloseDriver() {
        driver.close();
    }

    @Override
    public void QuitDriver() {
        if (driver != null)
            driver.quit();
    }
}

File: DotNet/Framework/AutomationFramework/Helpers/Drivers/Chrome.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Remote;

namespace Helpers.Drivers
{
    public class Chrome : IDriver
    {
        IWebDriver driver;

        public void CloseDriver()
        {
            driver.Quit();
        }

        public IWebDriver GetDriver()
        {
            if (driver == null)
                SetDriver();

            return driver;
        }

        public void MaximizeDriver()
        {
            driver.Manage().Window.Maximize();
        }

        public void MinimizeDriver()
        {
            driver.Manage().Window.Minimize();
        }

        public void NavigateURL(string url)
        {
            driver.Navigate().GoToUrl(url);
        }

        public void SetDriver()
        {
            ChromeOptions chromeOptions = new ChromeOptions();
            chromeOptions.AddArgument("--disable-dev-shm-usage");
            chromeOptions.AddArgument("--no-sandbox");
//            chromeOptions.AddArgument("--headless");
            chromeOptions.AddArgument("ignore-certificate-errors");
            chromeOptions.PlatformName = "linux";
            this.driver = new RemoteWebDriver(new Uri("http://localhost:4444/wd/hub"), chromeOptions);
        }
    }
}
